aea/configurations/data_types.py

```diff
@@ -223,6 +223,8 @@ class PublicId(JSONSerializable):
     True
     """
 
+    __slots__ = ("_author", "_name", "_package_version")
```

Comment:
Adding slots in heavily used framework-only classes is sensible. We only need to be careful with classes which we want the user to be able to extend.